fsspec/dircache.py

```diff
@@ -93,6 +97,95 @@ def __iter__(self):
 
     def __reduce__(self):
         return (
-            DirCache,
+            MemDirCache,
             (self.use_listings_cache, self.listings_expiry_time, self.max_paths),
         )
+
+
+class FileDirCache(MutableMapping):
+    def __init__(
+        self,
+        use_listings_cache=True,
+        listings_expiry_time=None,
+        listings_cache_location=None,
+        **kwargs,
+    ):
+        """
+
+        Parameters
+        ----------
+        use_listings_cache: bool
+            If False, this cache never returns items, but always reports KeyError,
+            and setting items has no effect
+        listings_expiry_time: int or float (optional)
+            Time in seconds that a listing is considered valid. If None,
+            listings do not expire.
+        listings_cache_location: str (optional)
+            Directory path at which the listings cache file is stored. If None,
+            an autogenerated path at the user folder is created.
+
+        """
+        import platformdirs
+        from diskcache import Cache
+
+        listings_expiry_time = listings_expiry_time and float(listings_expiry_time)
+
+        if listings_cache_location:
+            listings_cache_location = Path(listings_cache_location) / str(
```

Comment:
Why would we use Path within a filesystem library :)

Reply:
Can diskcache handle fsspec paths? E.g., memory://cache would be a very similar solution to MemDirCache. Such an option might even be useful for semi-speedy remotes like smb: